src/clr-core/button/button.element.ts

```diff
@@ -94,48 +134,67 @@ export class CdsButton extends CdsBaseButton {
     }
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (this.icon) {
+      this.icon.setAttribute('slot', 'button-icon');
+    }
+
+    if (this.badge) {
+      this.badge.setAttribute('slot', 'button-badge');
+    }
+  }
+
   update(props: Map<string, any>) {
     if (this.privateHost && props.has('loadingState')) {
       this.updateLoadingState();
     }
     super.update(props);
   }
 
-  // nested span tags allow for line-height erasers on the innermost span and flex-based centering on the outermost span
   render() {
-    return html`
-    <div class="private-host">
-      ${this.loadingState === ClrLoadingState.LOADING ? iconSpinner : ''}
-      ${this.loadingState === ClrLoadingState.SUCCESS ? iconSpinnerCheck : ''}
-      ${this.loadingState === ClrLoadingState.DEFAULT ? html`<span><span><slot></slot></span></span>` : ''}
-      ${this.hiddenButtonTemplate}
-    </div>
-    `;
+    const loadingState = this.loadingState;
+    const hasIcon = !!this.icon;
+    const hasBadge = !!this.badge;
+
+    return html`<div class="private-host" cds-layout="horizontal">${
+      loadingState === ClrLoadingState.SUCCESS ? html`${iconSpinnerCheck}` : ''
+    }${loadingState === ClrLoadingState.ERROR ? html`${iconSpinnerError}` : ''}${
+      loadingState === ClrLoadingState.LOADING ? html`${iconSpinner}` : ''
+    }${loadingState === ClrLoadingState.DEFAULT ? html`${buttonSlots(hasIcon, hasBadge)}` : ''}${
+      this.hiddenButtonTemplate
+    }</div>`;
   }
 
   static get styles() {
-    return [baseStyles, styles];
+    return [baseStyles, baseButtonStyles, styles];
   }
 
   private updateLoadingState() {
-    if (this.loadingState === ClrLoadingState.LOADING) {
-      this.disableButton();
-    } else if (this.loadingState === ClrLoadingState.SUCCESS) {
-      this.disableButton();
-      setTimeout(() => this.enableButton(), 1000);
-    } else {
-      this.enableButton();
+    switch (this.loadingState) {
+      case ClrLoadingState.LOADING:
```

Comment:
I kept each of the loadingStates as their own branch in the switch statement in the event that we need more logic/behavior associated with them. Think animations and such.